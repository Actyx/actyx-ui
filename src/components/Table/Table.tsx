import { theme } from '../../theme';
import * as classNames from 'classnames';
import * as React from 'react';
import injectSheet, { StyleSheet, WithStyles } from 'react-jss';
import { compose, setDisplayName } from 'recompose';

type CompProps = Readonly<{
  className?: string;
  children: React.ReactNode;
  alternateColor?: boolean;
  dark?: boolean;
}>;

type Props = WithStyles<ClassKey> & CompProps;

const TableComp = ({ className, classes, children, alternateColor, dark }: Props) => (
  <table
    className={classNames(
      {
        [classes.tableDark]: dark,
        [classes.tableLight]: !dark,
        [classes.rowBorderDark]: dark,
        [classes.rowBorder]: !dark,
        [classes.alternateColorDark]: dark && alternateColor,
        [classes.alternateColor]: !dark && alternateColor
      },
      className
    )}
  >
    {children}
  </table>
);

type ClassKey =
  | 'tableLight'
  | 'tableDark'
  | 'alternateColor'
  | 'alternateColorDark'
  | 'rowBorder'
  | 'rowBorderDark';

const styles: StyleSheet<ClassKey, CompProps> = {
  tableLight: {
    width: '100%',
    borderCollapse: 'collapse',
    border: 'none',
    backgroundColor: theme.palette.common.white
  },
  tableDark: {
    width: '100%',
    borderCollapse: 'collapse',
    border: 'none',
    backgroundColor: theme.palette.grey[900]
  },
  alternateColor: {
    '& tbody > tr:nth-of-type(odd)': {
      backgroundColor: theme.palette.grey[50]
    }
  },
  alternateColorDark: {
    '& tbody > tr:nth-of-type(odd)': {
      backgroundColor: theme.palette.grey[700]
    }
  },
  rowBorder: {
    '& thead > tr': {
      borderBottom: `${1}px solid ${theme.palette.grey[400]}`
    },
    '& tbody > tr': {
      borderBottom: `${1}px solid ${theme.palette.grey[400]}`
    }
  },
  rowBorderDark: {
    '& thead > tr': {
      borderBottom: `${1}px solid ${theme.palette.grey[400]}`
    },
    '& tbody > tr': {
      borderBottom: `${1}px solid ${theme.palette.grey[900]}`
    }
  }
};

export const Table = compose<Props, CompProps>(
  setDisplayName('Table'),
  injectSheet(styles)
)(TableComp);
