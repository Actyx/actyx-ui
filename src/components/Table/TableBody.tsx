import * as React from 'react';
import injectSheet, { StyleSheet, WithStyles } from 'react-jss';
import { compose, setDisplayName } from 'recompose';

type CompProps = Readonly<{
  children: React.ReactNode;
}>;

type Props = WithStyles<ClassKey> & CompProps;

const TableBodyComp = ({ classes, children }: Props) => (
  <tbody className={classes.tableBody}>{children}</tbody>
);

type ClassKey = 'tableBody';

const styles: StyleSheet<ClassKey, CompProps> = {
  tableBody: {}
};

export const TableBody = compose<Props, CompProps>(
  setDisplayName('TableBody'),
  injectSheet(styles)
)(TableBodyComp);
