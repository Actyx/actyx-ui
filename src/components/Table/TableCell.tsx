import * as classNames from 'classnames';
import * as React from 'react';
import injectSheet, { StyleSheet, WithStyles } from 'react-jss';
import { compose, setDisplayName } from 'recompose';

type CompProps = Readonly<{
  className?: string;
  children: React.ReactNode;
  numeric?: boolean;
  style?: React.CSSProperties;
  cellType?: string;
}>;

type ClassKey = 'tableCell' | 'numeric';

type Props = WithStyles<ClassKey> & CompProps;

const TableCellComp = ({ classes, className, children, numeric, style, cellType }: Props) => (
  <td
    data-celltype={cellType}
    className={classNames(classes.tableCell, { [classes.numeric]: numeric }, className)}
    style={style}
  >
    {children}
  </td>
);

const styles: StyleSheet<ClassKey> = {
  tableCell: {
    padding: 10
  },
  numeric: {
    textAlign: 'right'
  }
};

export const TableCell = compose<Props, CompProps>(
  setDisplayName('TableCell'),
  injectSheet(styles)
)(TableCellComp);
