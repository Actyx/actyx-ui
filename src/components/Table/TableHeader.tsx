/*
 * Copyright 2020 Actyx AG
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from 'react';
import injectSheet, { StyleSheet, WithStyles } from 'react-jss';
import { compose, setDisplayName } from 'recompose';

type CompProps = Readonly<{
  children: React.ReactNode;
}>;

type Props = WithStyles<ClassKey> & CompProps;

const TableHeaderComp = ({ classes, children }: Props) => (
  <thead className={classes.thead}>{children}</thead>
);

type ClassKey = 'thead';

const styles: StyleSheet<ClassKey> = {
  thead: {
    '& tr': {
      height: 40
    }
  }
};

export const TableHeader = compose<Props, CompProps>(
  setDisplayName('TableHeader'),
  injectSheet(styles)
)(TableHeaderComp);
