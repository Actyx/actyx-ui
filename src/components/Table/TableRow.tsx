/*
 * Copyright 2020 Actyx AG
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { theme } from '../../theme';
import * as classNames from 'classnames';
import * as React from 'react';
import injectSheet, { StyleSheet, WithStyles } from 'react-jss';
import { compose, setDisplayName } from 'recompose';

type CompProps = Readonly<{
  className?: string;
  children: React.ReactNode;
  active?: boolean;
  condensed?: boolean;
  onSelect?: () => void;
}>;

type Props = WithStyles<ClassKey> & CompProps;

const TableRowComp = ({ classes, className, children, active, condensed, onSelect }: Props) => (
  <tr
    onClick={onSelect}
    className={classNames(condensed === true ? classes.condensed : classes.tableRow, className)}
    style={active ? { backgroundColor: theme.palette.primary.light } : {}}
  >
    {children}
  </tr>
);

type ClassKey = 'tableRow' | 'condensed';

const styles: StyleSheet<ClassKey> = {
  tableRow: {
    height: 80
  },
  condensed: {
    height: 60
  }
};

export const TableRow = compose<Props, CompProps>(
  setDisplayName('TableRow'),
  injectSheet(styles)
)(TableRowComp);
