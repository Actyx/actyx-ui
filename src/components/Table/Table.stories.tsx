import { storiesOf } from '@storybook/react';
import { theme } from '../../theme';
import { hostDecorator } from '../../utils';
import * as React from 'react';
import { Typography, Variant } from '../Typography';
import { Table } from './Table';
import { TableBody } from './TableBody';
import { TableCell } from './TableCell';
import { TableHeader } from './TableHeader';
import { TableRow } from './TableRow';

const COLOR_WHITE = theme.palette.grey.white;

const createTableCell = (
  children: React.ReactNode,
  variant: Variant,
  dark: boolean,
  numeric = false,
  idx: number
) => (
  <TableCell key={idx} numeric={numeric}>
    <Typography variant={variant} color={dark ? COLOR_WHITE : undefined}>
      {children}
    </Typography>
  </TableCell>
);

type DataTable = ReadonlyArray<ReadonlyArray<string>>;

const dataHeader: DataTable = [['Description', 'User', 'Status', 'Amount']];

const dataBody: DataTable = [
  ['Machine Cleaning: ABC 1 of 20', 'Johanna Maltese', 'Finished', '10'],
  ['Machine Cleaning: XYZ 10 of 30', 'Robert Parker', 'Finished', '10'],
  ['Machine Cleaning: QWE 5 of 5', 'Linda Berry', 'Finished', '20'],
  ['Machine Cleaning: JKL 1 of 2', 'Abram Green', 'Finished', '10'],
  ['Machine Cleaning: BNM 2 of 3', 'James Brown', 'Finished', '20'],
  ['Machine Cleaning: PQR 1 of 3', 'Tony Tesla', 'Finished', '10']
];

const createRows = (dataRow: DataTable, variant: Variant = 'standard', dark = false) => (
  <>
    {dataRow.map((row, rIdx: number) => (
      <TableRow key={rIdx} active={rIdx === 1}>
        {row.map((cell, idx: number) => createTableCell(cell, variant, dark, idx === 3, idx))}
      </TableRow>
    ))}
  </>
);

storiesOf('common|Table.Table', module)
  .addDecorator(
    hostDecorator({
      width: 900,
      backgroundColor: theme.palette.grey.dark400
    })
  )
  .add('body', () => (
    <Table>
      <TableBody>{createRows(dataBody)}</TableBody>
    </Table>
  ))
  .add('header body', () => (
    <Table alternateColor>
      <TableHeader>{createRows(dataHeader, 'subtext')}</TableHeader>
      <TableBody>{createRows(dataBody)}</TableBody>
    </Table>
  ))
  .add('header body alternateColor dark', () => (
    <Table alternateColor dark>
      <TableHeader>{createRows(dataHeader, 'subtext', true)}</TableHeader>
      <TableBody>{createRows(dataBody, undefined, true)}</TableBody>
    </Table>
  ));
