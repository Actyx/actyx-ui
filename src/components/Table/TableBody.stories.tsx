import { storiesOf } from '@storybook/react';
import { Typography } from '../Typography';
import { hostDecorator } from '../../utils';
import * as React from 'react';
import { Table } from './Table';
import { TableBody } from './TableBody';
import { TableCell } from './TableCell';
import { TableRow } from './TableRow';

storiesOf('Components|Table.TableBody', module)
  .addDecorator(
    hostDecorator({
      width: 900
    })
  )
  .add('base', () => (
    <Table>
      <TableBody>
        <TableRow>
          <TableCell>
            <Typography variant="standard">Machine Cleaning: ABC 1 of 20</Typography>
          </TableCell>
          <TableCell>
            <Typography variant="standard">Johanna Maltese</Typography>
          </TableCell>
          <TableCell>
            <Typography variant="standard">Finished</Typography>
          </TableCell>
          <TableCell numeric>
            <Typography variant="standard">10</Typography>
          </TableCell>
        </TableRow>
      </TableBody>
    </Table>
  ));
