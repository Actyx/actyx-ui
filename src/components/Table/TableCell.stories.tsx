import { storiesOf } from '@storybook/react';
import { Typography } from '../Typography';
import { hostDecorator } from '../../utils';
import * as React from 'react';
import { Table } from './Table';
import { TableBody } from './TableBody';
import { TableCell } from './TableCell';
import { TableRow } from './TableRow';

storiesOf('Components|Table.TableCell', module)
  .addDecorator(hostDecorator())
  .add('base', () => (
    <Table>
      <TableBody>
        <TableRow>
          <TableCell>
            <Typography variant="standard">10</Typography>
          </TableCell>
        </TableRow>
      </TableBody>
    </Table>
  ))
  .add('numeric', () => (
    <Table>
      <TableBody>
        <TableRow>
          <TableCell numeric>
            <Typography variant="standard">10</Typography>
          </TableCell>
        </TableRow>
      </TableBody>
    </Table>
  ))
  .add('cellType', () => (
    <Table>
      <TableBody>
        <TableRow>
          <TableCell numeric cellType={'cellType'}>
            <Typography variant="standard">10</Typography>
          </TableCell>
        </TableRow>
      </TableBody>
    </Table>
  ));
