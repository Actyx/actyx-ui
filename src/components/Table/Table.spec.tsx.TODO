import { shallow } from 'enzyme';
import * as React from 'react';
import { Table } from './Table';
import { TableBody } from './TableBody';
import { TableCell } from './TableCell';
import { TableHeader } from './TableHeader';
import { TableRow } from './TableRow';

const render = () => (
  <>
    <TableHeader>
      <TableRow>
        <TableCell>Header</TableCell>
      </TableRow>
    </TableHeader>
    <TableBody>
      <TableRow>
        <TableCell>Body</TableCell>
      </TableRow>
    </TableBody>
  </>
);

describe('<Table />', () => {
  it('should render', () => {
    const wrapper = shallow(<Table>{render()}</Table>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render alternateColor', () => {
    const wrapper = shallow(<Table alternateColor>{render()}</Table>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render dark', () => {
    const wrapper = shallow(<Table dark>{render()}</Table>);
    expect(wrapper).toMatchSnapshot();
  });
});
