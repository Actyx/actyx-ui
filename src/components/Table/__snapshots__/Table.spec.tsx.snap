// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render 1`] = `ShallowWrapper {}`;

exports[`<Table /> should render alternateColor 1`] = `ShallowWrapper {}`;

exports[`<Table /> should render dark 1`] = `ShallowWrapper {}`;
